### PR TITLE
test: add addon to test-app

### PR DESCRIPTION
**Pull request by danbev**
This commit adds a native addon to the test-app to verify that the s2i
image can handle it.

### CODE REVIEW

**Comment by helio-frota** [comment]
LGTM :+1:

File: test/test-app/package.json [hunk 1]
```diff
@@ -6,15 +6,18 @@
   "license": "MIT",
   "scripts": {
     "dev": "nodemon --ignore node_modules/ server.js",
-    "postinstall": "tsc greeting.ts",
-    "start": "echo 'starting node app' && node server.js"
+    "postinstall": "tsc greeting.ts && node-gyp -C addon configure build",
+    "start": "echo 'starting node app' && node server.js",
+    "compile": "node-gyp -C addon configure build"
```
Comment (lance) [inline test/test-app/package.json]: @danbev you've added a script target here, but it's not used. Did you mean for the `postinstall` to be `"tsc greeting.ts && npm run compile"`?

**Comment by lance** [comment]
Thanks for doing this Dan!

File: test/test-app/package.json (hunk 1 above)
Comment (danbev) [reply test/test-app/package.json]: Sorry, that was just a left over. I'll remove it.

**Pull request merged**